ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -657,6 +666,14 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
                 << errno
                 << "(" << strerror(errno) << ")"
                 << endl;
+            string errorMessage = ss.str();
+
+            s_callbacks.LogError(errorMessage.c_str());
+            cout << errorMessage << endl;
+
+            free(messageMemory);
+            result = PrjFS_Result_EPathNotFound;
```

Comment:
We need to call `SendKernelMessageResponse(requestHeader->messageId, responseType);` with this result or the kext will hang waiting on our response.

Rather than an early return adding a `goto` and a `label` (where we do this below) might be the cleaner options.

Reply:
Good catch! Make sure we don't double-free the `messageMemory`. (One of these days I want to come up with a good RAII way of handling that memory, but that's not for now.)

Reply:
> with this result

Actually, it looks like we're currently mapping the result to Success/Failure (see below).

#183 is open for sending back the the more detailed error code.

Reply:
`PrjFS_Result_EPathNotFound`

@jeschu1 before merging, can you check with @nickgra how returning either success or failure in this situation impacts his scenario?

I'm wondering if we have two requests racing for to delete the same file, will returning Failure here cause the second delete to fail when perhaps it would have otherwise succeeded (or resulted in an error of "No such file or directory")?

If anything this might raise the importance of #183 as we'd be able to return a more describe error code here.

cc: @pmj for his thoughts on this as well.

Reply:
@wilbaker , this is a good question.

I guess whats the harm in us returning success?  We'll have the entry in logs / telemetry so we don't really need to fail.

Reply:
We talked offline and decided always returning success is the best approach for now.

Reply:
@nickgra actually maybe start running tests where we return success?  I think we still want both data points thought.